### ISSUE TITLE
ERROR: Not implemented endpoint

### ISSUE THREAD

**Issue by unzap** [conan-io/conan#4914]
Artifactory Community Edition for C/C++
6.8.2 rev 60802900

I have Artifactory running on my local computer. I have a few repositories created containing Conan packages. I can search and install packages successfully from the other local Artifactory repositories.

I've created a virtual repository to collect all the other actual repositories in it.
Include filter is "**/*" and no exclude filter used. Repository type is "generic" and layout is "conan-default". In the "included repositories" list I can see all my actual repositories.

`conan search -r=some-repo-in-artifactory`
`... # lists packages ok`
`conan remote add virtual-test http://localhost:8081/artifactory/api/conan/virtual-test`
`conan search -r=virtual-test`
`ERROR: Not implemented endpoint. [Remote: virtual-test]`

I've tried to search through the Artifactory documentation and double-checked the virtual repository settings in Artifactory but haven't been able to figure out what is wrong still in my setup?

**Comment by unzap** [comment]
From the server log I can see the following:

>2019-04-05 12:19:49,532 [http-nio-8081-exec-3] [WARN ] (o.a.a.c.r.ConanResource:83) - Local repository 'virtual-test' not found

**Comment by lasote** [comment]
The virtual repository support is not ready yet for Conan repositories. You might try with Artifactory 6.9, I think there might be enabled but still not visible in the UI (only rest-api), but never with generic repositories, because Conan only knows how to talk the "conan protocol"

**Comment by unzap** [comment]
Hi @lasote 

> The virtual repository support is not ready yet for Conan repositories. You might try with Artifactory 6.9, I think there might be enabled but still not visible in the UI (only rest-api), but never with generic repositories, because Conan only knows how to talk the "conan protocol"

Thanks for the information. Do you know when the official support should be released?

**Comment by lasote** [comment]
Probably next Artifactory release, but I don't know to be honest. What I know is that it will work only with the [revisions](https://docs.conan.io/en/latest/mastering/revisions.html) mechanism enabled.

**Comment by unzap** [comment]
I downloaded 6.9 and tried to get started (rest API) but now I get similar error with regular generic repository as well?

ERROR: Not implemented endpoint. [Remote: artifactory-generic-local]

This is generic repository with "conan-default" layout? I noticed the "Conan" package type was greyed out in the CE version. Should I download evaluation version of the Pro to get the Conan package type, it this has anything to do with the error?

**Comment by lasote** [comment]
Remotes and Virtuals are not supported in the CE version. I think that might be the error.

**Comment by unzap** [comment]
@lasote hmm, it did work with the previous version of the Artifactory CE (6.8.2)? With 6.8.2 I was able to create local repositories (conan layout) and push/download packages from the repositories?

I updated to 6.9 and similarly configured a generic repo (conan layout), but it gives that error?

**Comment by lasote** [comment]
A generic repo with Conan layout worked previously? Not directly a Conan repository? Using the conan client?

**Comment by unzap** [comment]
[https://pasteboard.co/I9j01fD.png](https://pasteboard.co/I9iZHi6.png)

[https://pasteboard.co/I9j01fD.png](https://pasteboard.co/I9j01fD.png)

In $HOME/.conan/registry.json:

`  {`
`   "name": "artifactory-generic-local",`
`   "url": "http://localhost:8081/artifactory/api/conan/generic-local",`
`   "verify_ssl": true`
`  }`

`$ conan search -r=artifactory-generic-local`
`ERROR: Not implemented endpoint. [Remote: artifactory-generic-local]`

Has something changed since 6.8.2, am I configuring the local repository wrongly as identical configuration did work with 6.8.2?

**Comment by lasote** [comment]
I will report to Artifactory support, I had no idea about that working. The conan client is meant to work only with a Conan repository.

**Comment by unzap** [comment]
@lasote oh then I got lucky with the older version? I wanted to upgrade because I want to test virtual repository support for Conan repositories.

I've registered for evaluation of the pro version now so I should be able to continue..

**Comment by unzap** [comment]
I was able to create Virtual repository of type Conan using the Rest API. After the creation I can see "Conan" icon in the virtual repository configuration.
But searching from the virtual repository still fails though:

`ERROR: {`
`  "errors" : [ {`
`    "status" : 400,`
`    "message" : "Unsupported Conan v1 repository request for 'virtual-test3'"`
`  } ]`
`}. [Remote: artifactory-pro-virtual-test3]`

I guess I need to wait until some future Artifactory release to have Virtual repository support for Conan repositories?

**Comment by lasote** [comment]
Have you activated the revisions in the client? Check here how to do it: https://docs.conan.io/en/latest/mastering/revisions.html#how-to-activate-the-revisions

**Comment by unzap** [comment]
@lasote yes, now it works! Thanks for pointing out about 'revisions' feature :)
